src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/EventPipeSession.cs

```diff
@@ -30,16 +29,84 @@ private EventPipeSession(IpcEndpoint endpoint, IpcResponse response, ulong sessi
 
         internal static EventPipeSession Start(IpcEndpoint endpoint, EventPipeSessionConfiguration config)
         {
-            IpcMessage requestMessage = CreateStartMessage(config);
+            IpcMessage requestMessage = CreateStartMessage(config, EventPipeCommandId.CollectTracing4);
             IpcResponse? response = IpcClient.SendMessageGetContinuation(endpoint, requestMessage);
-            return CreateSessionFromResponse(endpoint, ref response, nameof(Start));
+            try
+            {
+                EventPipeSession session = null;
+                if (DiagnosticsClient.ValidateResponseMessage(response.Value.Message, nameof(StartAsync), DiagnosticsClient.ValidateResponseOptions.UnknownCommandReturnsFalse))
+                {
+                    session = CreateSessionFromResponse(endpoint, ref response);
+                    // The session owns the continuation stream
+                    response = null;
+                    return session;
+                }
+
+                requestMessage = CreateStartMessage(config, EventPipeCommandId.CollectTracing3);
+                response = IpcClient.SendMessageGetContinuation(endpoint, requestMessage);
+
+                if (DiagnosticsClient.ValidateResponseMessage(response.Value.Message, nameof(StartAsync), DiagnosticsClient.ValidateResponseOptions.UnknownCommandReturnsFalse))
+                {
+                    session = CreateSessionFromResponse(endpoint, ref response);
+                    // The session owns the continuation stream
+                    response = null;
+                    return session;
+                }
+
+                requestMessage = CreateStartMessage(config, EventPipeCommandId.CollectTracing2);
```

Comment:
Falling back to `CollectTracing2` here may not be the right thing to do here if having stack is fatal (e.g. PII concern)